src/Workspaces/Core/Portable/FindSymbols/FindReferences/Finders/AbstractReferenceFinder_GlobalSuppressions.cs

```diff
@@ -0,0 +1,298 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+using System.Collections.Immutable;
+using System.Diagnostics.CodeAnalysis;
+using System.Text.RegularExpressions;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.LanguageServices;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.FindSymbols.Finders
+{
+    internal abstract partial class AbstractReferenceFinder : IReferenceFinder
+    {
+        /// <summary>
+        /// Regex for symbol documentation comment ID.
+        /// For example, "~M:C.X(System.String)" represents the documentation comment ID of a method named 'X'
+        /// that takes a single string typed parameter and is contained in a type named 'C'.
+        /// 
+        /// We divide the regex it into 3 groups:
+        /// 1. Prefix:
+        ///     - Starts with an optional '~'
+        ///     - Followed by a single capital letter indicating the symbol kind (for example, 'M' indicates method symbol)
+        ///     - Followed by ':'
+        /// 2. Core symbol ID, which is its fully qualified name before the optional parameter list and return type (i.e. before the '(' or '[' tokens)
+        /// 3. Optional parameter list and/or return type that begins with a '(' or '[' tokens.
+        /// 
+        /// For the above example, "~M:" is the prefix, "C.X" is the core symbol ID and "(System.String)" is the parameter list.
+        /// </summary>
+        private static readonly Regex s_docCommentIdPattern = new Regex(@"(~?[A-Z]:)([^([]*)(.*)");
```

Comment:
do we have tests where the pattern doesn't match this?  (just to test hte negative path)?

Reply:
Sure, I have added it to https://github.com/dotnet/roslyn/issues/44401. I will add the test it in the next PR when I handle references in parameter types/return types.